project_milestone_status/models/project_milestone.py

```diff
@@ -0,0 +1,40 @@
+from odoo import api, fields, models
+
+
+class ProjectMilestone(models.Model):
+    _inherit = "project.milestone"
+
+    execution = fields.Integer(compute="_compute_execution")
+    dedication = fields.Integer(compute="_compute_dedication")
+
+    @api.depends("task_ids")
+    def _compute_execution(self):
+        for milestone in self:
+            fold_stage_ids = milestone.env["project.task.type"].search(
```

Comment:
Same remark as above. For performances reasons, you should not do a `search()` in a loop!

```suggestion
            executed_tasks = milestone.task_ids.filtered("fold_stage_ids.fold")
```